hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenMintHandler.java

```diff
@@ -202,6 +202,7 @@ private List<Long> mintNonFungible(
         final var copyTreasury = treasuryAccount.copyBuilder();
         // Update Token and treasury
         copyToken.lastUsedSerialNumber(currentSerialNumber);
+        copyToken.totalSupply(token.totalSupply() + metadataCount);
```

Comment:
This should be done on line 185 right ? 🤔 I must be missing something.

Reply:
Exactly, but not sure why it's not persisted. My theory, could be wrong, is that where we use again tokenStore to put the new token we overwrite the previously saved object from line 185 ?

Reply:
The token is modified in `changeSupply`, but we are not fetching it again from `tokenStore` before modifying it 🙂 . Fixed and pushed change

Reply:
Thanks @georgi-l95